### PR TITLE
[5733] Renaming items or editing Page URL via UI, returns HTTP 404 on /content/item_by_path call

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5733

### CODE REVIEW

File: ui/app/src/components/CreateFolderDialog/CreateFolderContainer.tsx [hunk 1]
```diff
@@ -64,6 +64,12 @@ export function CreateFolderContainer(props: CreateFolderContainerProps) {
     : itemLookupTable[newFolderPath] !== UNDEFINED;
   const isValid = !isBlank(name) && !folderExists && (!rename || name !== value);
 
+  // If folder name changes, useDetailedItem will try to re-fetch a non-existing item (old path), so we will only re-fetch
+  // if path changes
```
Comment (rart) [inline ui/app/src/components/CreateFolderDialog/CreateFolderContainer.tsx]: This comment is a bit confusing. If the folder name changes the path changes so whether it is useDetailed or this effect, they would both react to that no? What was the idea here? The comment needs to state 1. not to use useDetailedItem and why 2. why/how the effect solves.

File: ui/app/src/components/LegacyFormDialog/EmbeddedLegacyContainer.tsx [hunk 2]
```diff
@@ -79,9 +80,16 @@ export const EmbeddedLegacyContainer = React.forwardRef(function EmbeddedLegacyE
   const iframeRef = useRef(null);
   const dispatch = useDispatch();
   const [error, setError] = useState<ApiResponse>(null);
-  const item = useDetailedItem(path);
+  const [item, setItem] = useState(null);
+  const siteId = useActiveSiteId();
   const availableActions = item?.availableActions;
 
+  // If filename changes, useDetailedItem will try to re-fetch a non-existing item (old path), so we will only re-fetch
+  // if path changes
```
Comment (rart) [inline ui/app/src/components/LegacyFormDialog/EmbeddedLegacyContainer.tsx]: As above. 

Also, is it worth creating a hook for both of these instances?

**Review by rart — CHANGES_REQUESTED** [review]

**Comment by jvega190** [comment]
@rart I updated the comments, and created a hook called useDetailedItemNoState

File: ui/app/src/hooks/useDetailedItemNoState.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export function useDetailedItemNoState(path: string): DetailedItem {
+  const [item, setItem] = useState(null);
+  const siteId = useActiveSiteId();
+
+  useEffect(() => {
+    fetchDetailedItem(siteId, path).subscribe(setItem);
+  }, [siteId, path]);
+
+  return item;
+}
```
Comment (rart) [inline ui/app/src/hooks/useDetailedItemNoState.ts]:
```suggestion
}

export default useDetailedItemNoState;

```

**Review by rart — CHANGES_REQUESTED** [review]
Do we actually need a **detailed** item here? It's a bit more expensive to fetch so if Sandbox does, we should go for that.

As far as the new hook, lets try something like....
```ts
export function useFetchItem(path: string, detailed: boolean = false): DetailedItem {
   const [item, setItem] = useState(null);
   const siteId = useActiveSiteId();
   useEffect(() => {
     if (detailed) 
       fetchDetailedItem(siteId, path).subscribe(setItem);
      else
       fetchSandboxItem(sideId, path, { caseAsDetailedItem: true }).subscribe(setItem)
   }, [siteId, path, detailed]);
   return item;
 }
```

File: ui/app/src/components/RenameAssetDialog/RenameAssetDialogContainer.tsx [hunk 4]
```diff
@@ -144,7 +144,7 @@ export function RenameAssetDialogContainer(props: RenameAssetContainerProps) {
         <form
           onSubmit={(e) => {
             e.preventDefault();
-            if (renameDisabled) {
+            if (!renameDisabled) {
```
Comment (jvega190) [inline ui/app/src/components/RenameAssetDialog/RenameAssetDialogContainer.tsx:147]: Pushed this as a small fix (issue preventing submit on enter)

File: ui/app/src/hooks/useFetchItem.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  const siteId = useActiveSiteId();
+  useEffect(() => {
+    if (detailed) {
+      fetchDetailedItem(siteId, path).subscribe(setItem);
+    } else {
+      fetchSandboxItem(siteId, path, { castAsDetailedItem: true }).subscribe(setItem);
+    }
+  }, [siteId, path, detailed]);
+  return item;
+}
```
Comment (rart) [inline ui/app/src/hooks/useFetchItem.ts:33]:
```suggestion
}

export default useFetchItem;

```

File: ui/app/src/hooks/useFetchItem.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect, useState } from 'react';
+import useActiveSiteId from './useActiveSiteId';
+import { fetchDetailedItem, fetchSandboxItem } from '../services/content';
+import { DetailedItem } from '../models';
```
Comment (rart) [inline ui/app/src/hooks/useFetchItem.ts]:
```suggestion
import { DetailedItem } from '../models/Item';
```

**Review by rart — APPROVED** [review]

**Pull request merged**